### PR TITLE
Csp 1421 validate paye

### PR DESCRIPTION
**Pull request by MarkFCain**

### CODE REVIEW

File: src/ProviderPR/SFA.DAS.ProviderPR/InnerApi/Responses/GetRequestResponse.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using SFA.DAS.SharedOuterApi.Models.ProviderRelationships;
+
+namespace SFA.DAS.ProviderPR.InnerApi.Responses;
+public class GetRequestResponse
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ProviderPR/SFA.DAS.ProviderPR/InnerApi/Responses/GetRequestResponse.cs:4]: Not sure what you're asking here.  This object will, I think, be returned in 2 ways, which I think is what you are asking about. By a GET via the requestId (in a different branch), and a GET via the UKPRN and PayeRef (as implemented in this story).  IIRR I lifted this wholesale from one of Shaun's branches, so I am using his structure.

File: src/Shared/SFA.DAS.SharedOuterApi/Services/PensionRegulatorApiClient.cs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        return _apiClient.GetResponseCode(request);
+    }
+
+    public Task<ApiResponse<TResponse>> GetWithResponseCode<TResponse>(IGetApiRequest request)
+    {
+        return _apiClient.GetWithResponseCode<TResponse>(request);
+    }
+
+    public Task<IEnumerable<TResponse>> GetAll<TResponse>(IGetAllApiRequest request)
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/Shared/SFA.DAS.SharedOuterApi/Services/PensionRegulatorApiClient.cs:37]: We created it and are only using the method needed in our story. It is in the shared space.  These methods have to be implemented as we inherit from IInternalApiClient. If anyone reuses and needs a different method, they are free to add.

File: src/ProviderPR/SFA.DAS.ProviderPR/Application/Queries/GetRelationshipsByUkprnPayeAorn/GetRelationshipByUkprnPayeAornQueryHandler.cs [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+                queryResult.AccountLegalEntityId.Value, cancellationToken);
+
+        if (relationshipResponse.ResponseMessage.StatusCode == HttpStatusCode.OK)
+        {
+            queryResult.HasRelationship = true;
+            queryResult.Operations = relationshipResponse.GetContent().Operations.ToList();
+        }
+
+        return queryResult;
+    }
```
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ProviderPR/SFA.DAS.ProviderPR/Application/Queries/GetRelationshipsByUkprnPayeAorn/GetRelationshipByUkprnPayeAornQueryHandler.cs:89]: I have made some improvements, but it becomes equally unreadable if you have to pass through all the various objects in the calls, so I have made it a little more readable by wrapping the augementation of queryResult in methods where it's more than just setting one value.

Also, as per our discussion about error handling, I have put a throw invalidOperationException around the call to the pension regulator whenever it's not a OK or NotFound).  The other endpoints throw exceptions back elegantly.

File: src/ProviderPR/SFA.DAS.ProviderPR/InnerApi/Responses/GetRequestResponse.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MarkFCain) [reply src/ProviderPR/SFA.DAS.ProviderPR/InnerApi/Responses/GetRequestResponse.cs:4]: Shaun merged the story with the GetRequestResponse, so I have now created a new distinct response that inherits from it, and makes it clear what the response is specifically for

[the remark this answers is not in the document]
Comment (PeterDavidWilliams) [reply src/ProviderPR/SFA.DAS.ProviderPR/InnerApi/Responses/GetRequestResponse.cs:4]: Ok, happy with this

**Review by PeterDavidWilliams — APPROVED** [review]

**Pull request merged**